### PR TITLE
Add YCSB for Couchbase -- Child of PR #43; merge that one first

### PR DESCRIPTION
**Pull request by dustinblack**
For issue #37 

Adding couchbase deployment

make operator and pod versions and urls variables changeable by CR

move default vars to defaults dir

making operator and pod versions and URLs user-configurable

### CODE REVIEW

**Comment by dustinblack** [comment]
Rebased on master post-merge of PR #43

**Comment by dustinblack** [comment]
> We should remove the k8s_log.py and the library directory for the time being.

done

**Comment by dustinblack** [comment]
Rebase on master after merging PR #44 and add the changes to the YCSB benchmark.

File: deploy/cluster_role.yaml [hunk 1]
```diff
@@ -12,3 +12,37 @@ rules:
   - clusterrolebindings
   verbs:
   - '*'
+- apiGroups:
```
Comment (aakarshg) [inline deploy/cluster_role.yaml]: seems to be repeated

File: roles/uperf-bench/library/k8s_log.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/python
```
Comment (aakarshg) [inline roles/uperf-bench/library/k8s_log.py]: This needs to be deleted

File: roles/ycsb-bench/meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+galaxy_info:
```
Comment (aakarshg) [inline roles/ycsb-bench/meta/main.yml]: This shouldn't be needed

**Comment by dustinblack** [comment]
Nits resolved

File: roles/ycsb-bench/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      apiVersion: v1
+      metadata:
+        name: '{{ meta.name }}-ycsb-bench'
+        namespace: '{{ meta.namespace }}'
+        labels:
+          name: 'ycsb-run'
+      spec:
+        containers:
+        - name: bench-server
+          image: "quay.io/dustinblack/ycsb-server:testing"
```
Comment (dustinblack) [inline roles/ycsb-bench/tasks/main.yml]: @aakarshg @jtaleric What is the right thing to do here? This image should be built and hosted alongside the operator image, shouldn't it?

Comment (jtaleric) [reply roles/ycsb-bench/tasks/main.yml]: @dustinblack I can give you access to the benchmark-operator namespace to push images to (i thought I already did)...

**Comment by dustinblack** [comment]
@jtaleric @aakarshg Please have another (final?) look at the PR. RBAC cleaned up; benchmark workloads changed from Pod to Job; leaving pods around after completion to manually scrape logs for output (until we have some kind of post-processing).

**Pull request merged**